### PR TITLE
Add doxygen into the g2_gbytesc.f

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Add doxygen into the g2_gbytesc.f

Part of #55

### CODE REVIEW

File: src/g2_gbytesc.f [hunk 1]
```diff
@@ -1,57 +1,82 @@
...
+
+!>    This subrountine is to extract arbitrary size values from a
+!>    packed bit string, right justifying each value in the unpacked
+!>    array without skip and interations.
+!>
+!>    @param[in] IN character*1 array input
+!>    @param[out] IOUT unpacked array output
+!>    @param[in] ISKIP initial number of bits to skip
+!>    @param[in] NBYTE number of bits to take
+
```
Comment (edwardhartnett) [inline src/g2_gbytesc.f:18]: Author tag missing.

File: src/g2_gbytesc.f [hunk 2]
```diff
@@ -1,57 +1,82 @@
...
 
+!>    This subrountine is to put arbitrary size values into a packed bit
+!>    string, taking the low order bits from each value in the unpacked
+!>    array without skip and interation.
+!>
+!>    @param[in] IN unpacked array input
+!>    @param[out] OUT packed array output
+!>    @param[in] ISKIP initial number of bits to skip
+!>    @param[in] NBYTE number of bits to pack
+
```
Comment (edwardhartnett) [inline src/g2_gbytesc.f:37]: Author tag missing.

File: src/g2_gbytesc.f [hunk 3]
```diff
@@ -1,57 +1,82 @@
...
+!>    packed bit string, right justifying each value in the unpacked
+!>    array with skip and interation options.
+!>
+!>    @param[in] IN character*1 array input
+!>    @param[out] IOUT unpacked array output
+!>    @param[in] ISKIP initial number of bits to skip
+!>    @param[in] NBYTE number of bits to take
+!>    @param[in] NSKIP additional number of bits to skip on each iteration
+!>    @param[in] N number of iterations
+
```
Comment (edwardhartnett) [inline src/g2_gbytesc.f:58]: Author tag missing.

File: src/g2_gbytesc.f [hunk 4]
```diff
@@ -63,24 +88,25 @@ SUBROUTINE G2_GBYTESC(IN,IOUT,ISKIP,NBYTE,NSKIP,N)
...
+!>    string, taking the low order bits from each value in the unpacked
+!>    array with skip and interation options.
+!>
+!>    @param[in] IN unpacked array input
+!>    @param[out] OUT packed array output
+!>    @param[in] ISKIP initial number of bits to skip
+!>    @param[in] NBYTE number of bits to pack
+!>    @param[in] NSKIP additional number of bits to skip on each iteration
+!>    @param[in] N number of iterations
+
```
Comment (edwardhartnett) [inline src/g2_gbytesc.f:113]: Author tag missing.

File: src/g2_gbytesc.f [hunk 5]
```diff
@@ -63,24 +88,25 @@ SUBROUTINE G2_GBYTESC(IN,IOUT,ISKIP,NBYTE,NSKIP,N)
...
+!>    This subrountine is to put arbitrary size values into a packed bit
+!>    string, taking the low order bits from each value in the unpacked
+!>    array with skip and interation options.
+!>
+!>    @param[in] IN unpacked array input
+!>    @param[out] OUT packed array output
+!>    @param[in] ISKIP initial number of bits to skip
+!>    @param[in] NBYTE number of bits to pack
+!>    @param[in] NSKIP additional number of bits to skip on each iteration
+!>    @param[in] N number of iterations
```
Comment (edwardhartnett) [inline src/g2_gbytesc.f]: Reorder these @param tags so the order is the same as the order of the parameters in the subroutine declaration.

File: src/g2_gbytesc.f [hunk 6]
```diff
@@ -1,57 +1,82 @@
+!>    @file
```
Comment (edwardhartnett) [inline src/g2_gbytesc.f]: @GeorgeGayno-NOAA am I correct in rembering that doxygen wants us to use C> for fixed format code? (That is, F77 code such as this?)

I seem to recall we discovered that in the UFS_UTILS doxygenation...

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/g2_gbytesc.f (hunk 5 above)
Comment (Hang-Lei-NOAA) [reply src/g2_gbytesc.f]: Okay, If author is unknown. I can add it based on the logical relationship. That should still be Stephen.

File: src/g2_gbytesc.f (hunk 6 above)
Comment (Hang-Lei-NOAA) [reply src/g2_gbytesc.f]: If so, I can change it back to C>.

File: src/g2_gbytesc.f (hunk 5 above)
Comment (edwardhartnett) [reply src/g2_gbytesc.f]: If you can make a good guess, do so.

If you can confirm with Stephen (or whomever you guess), do so.

If you can't even guess, put "NOAA programmer". This is a worst case scenario.

@DusanJovic-NOAA here's a great example of GitHub not providing the authorship info we want to have, and why NCEPLIBS code now required full attribution in the comments. This came up a lot in the UFS_UTILS doxygenation project - fortunately, there are still employees working here who remember who wrote what. Once those employees leave, then that information would be lost, had we not captured it in the documentation.

File: src/g2_gbytesc.f (hunk 6 above)
Comment (Hang-Lei-NOAA) [reply src/g2_gbytesc.f]: Let's wait for George's comments, and go further on this one.

Comment (edwardhartnett) [reply src/g2_gbytesc.f]: IIRC (and I may not) this manifested in doxygen warnings that did not go away until I changes the comments back to C>...

You can check the doxygen warnings in the CI build, and see if there are any warnings for this code. All warnings must be eliminated, so we can turn on automatic checking of documentation completeness in the CI system.

Comment (Hang-Lei-NOAA) [reply src/g2_gbytesc.f]: https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/g2__gbytesc_8f.html
This has been correctly shown on gh_pages

**Review by edwardhartnett — APPROVED** [review]

File: src/g2_gbytesc.f (hunk 6 above)
Comment (GeorgeGayno-NOAA) [reply src/g2_gbytesc.f]: @edwardhartnett The UFS_UTILS sfcsub.F routine is F77. To get it to work, I used the C> syntax at the top, but the !> syntax through the remainder of the file.  

See:  https://github.com/NOAA-EMC/UFS_UTILS/blob/develop/sorc/global_cycle.fd/sfcsub.F

**Review by edwardhartnett — COMMENTED** [review]
Looks good.

**Pull request merged**